File: sentimental-mario-less/mario.py


def main():
    # ask for block size
    while True:
        try:
            height = int(input("Height: "))
            if (height >= 1) and (height <= 8):
                break
        except:
            print("", end="")

    for i in range(height):
        space(height - i - 1)
        block(i + 1)
        print("")


def space(int):
    for i in range(int):
        print(" ", end="")


def block(int):
    for i in range(int):
        print("#", end="")


main()

File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

//ask for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 | height > 8);


//counter and vertical generator
    for (int printno = 1; printno - 1 < height; printno++)
    {
        //space generator
        for (int x = 0; x < height - printno; x++)
        {
            printf(" ");
        }
        //horizontal hash generator
        for (int x = 0; x < printno; x++)
        {
            printf("#");
        }

        printf("\n");
    }
}